src/go/writer/geometry_extended.go

```diff
@@ -13,14 +13,40 @@ func NewGeometryExtended(geometry *Geometry) *GeometryExtended {
 }
 
 func (g *GeometryExtended) BoundingBox() (minX, minY, maxX, maxY float64) {
+	if len(g.parts) == 0 {
+		return geomBoundingBox(g.Geometry)
+	}
+	minX = math.Inf(1)
+	minY = math.Inf(1)
+	maxX = math.Inf(-1)
+	maxY = math.Inf(-1)
+	for i := 0; i < len(g.parts); i++ {
+		partMinX, partMinY, partMaxX, partMaxY := geomBoundingBox(&g.parts[i])
+		if partMinX < minX {
+			minX = partMinX
+		}
+		if partMinY < minY {
+			minY = partMinY
+		}
+		if partMaxX > maxX {
+			maxX = partMaxX
+		}
+		if partMaxY > maxY {
+			maxY = partMaxY
+		}
+	}
+	return
+}
+
+func geomBoundingBox(g *Geometry) (minX, minY, maxX, maxY float64) {
 	firstPartEnd := len(g.xy)
 	if firstPartEnd == 0 {
 		// Nothing to do.
 		return
 	}
 
 	if len(g.ends) != 0 {
-		firstPartEnd = int(g.ends[0])
+		firstPartEnd = int(g.ends[0]) * 2
```

Comment:
Can you add a comment here to explain why you are multiplying by 2? It is not obvious form the code around it and it is also not clear to me why you need to do it (although I might be missing something).

Reply:
I added a comment. don't hesitate to tell me if it is not clear enough

Reply:
LGTM.